CardanoSharp.Wallet/CIPs/CIP2/Extensions/TransactionOutputExtensions.cs

```diff
@@ -42,17 +42,45 @@ public static Balance AggregateAssets(this IEnumerable<TransactionOutput> transa
                             balance.Assets.Add(nativeAsset);
                         }
 
-                        if(o.OutputPurpose == OutputPurpose.Burn) {
+                        nativeAsset.Quantity = nativeAsset.Quantity + na.Value;
+                    }
+                }
+            }
+
+            // remove / add assets from balance based on mint / burn token bundle
+            if (mint is not null) {

```

Comment:
A nit but I think an inverse condition `if (mint is null) return balance;` would help with the readability (less indents)